src/modules/acl/sagas.ts

```diff
@@ -36,22 +41,35 @@ function* handleFetchInfoRequest() {
 function* handleFetchWorldACLRequest(action: FetchWorldACLRequestAction) {
   const { targetContent, worldName } = action.payload
   try {
-    const acl: ACLResponse = yield call(getWorldACL, targetContent, worldName)
-    yield put(fetchWorldACLSuccess(acl))
+    const acl: WorldPermissionsResponse = yield call(getWorldACL, targetContent, worldName)
+    yield put(fetchWorldACLSuccess(acl, worldName))
   } catch (e) {
     yield put(fetchWorldACLFailure((e as any).message))
   }
 }
 
-function* handleUpdateWorldACLRequest(action: UpdateWorldACLRequestAction) {
+function* handlePutWorldACLRequest(action: PutWorldACLRequestAction) {
+  const { signature } = action.payload
+
+  const address: string = yield select(getAddress)
+
+  try {
+    yield call(updateWorldACL, { signature, address }, 'put')
+    yield put(putWorldACLSuccess())
+  } catch (e) {
+    yield put(putWorldACLFailure((e as any).message))
+  }
+}
+
+function* handleDeleteWorldACLRequest(action: DeleteWorldACLRequestAction) {
   const { signature } = action.payload
 
   const address: string = yield select(getAddress)
 
   try {
-    yield call(updateWorldACL, { signature, address })
-    yield put(updateWorldACLSuccess())
+    yield call(updateWorldACL, { signature, address }, 'delete')
+    yield put(deleteWorldACLSuccess())
   } catch (e) {
-    yield put(updateWorldACLFailure((e as any).message))
+    yield put(deleteWorldACLFailure((e as any).message))
```

Comment:
maybe you can merge these functions since you're doing the same and the `method` parameter has no use in `updateWorldACL`. Also, for the final users, it's just an update to their world's permissions